python/ray/serve/_private/http_state.py

```diff
@@ -195,10 +270,17 @@ def get_http_proxy_details(self) -> Dict[NodeId, HTTPProxyDetails]:
         }
 
     def update(self):
+        """Update the state of all HTTP proxies.
+
+        Start proxies on all nodes if not already exist and stop the proxies on nodes
+        that are no longer exist. Update all proxy states. Kill and restart
+        unhealthy proxies.
+        """
         self._start_proxies_if_needed()
         self._stop_proxies_if_needed()
         for proxy_state in self._proxy_states.values():
             proxy_state.update()
+        self._restart_unhealthy_proxies()
```

Comment:
I think it'd be preferable to have `start` and `stop` handle this internally (basically if something is unhealthy for long enough, it'll get stopped by `_stop_proxies_if_needed` and then `_start_proxies_if_needed` will notice and start a new one. That way there's exactly one place where start/stop logic is implemented.

This would also more closely match the deployment reconcile loop